src/submission/validation-clinical/therapy.ts

```diff
@@ -45,14 +45,101 @@ export const validate = async (
     throw new Error("Can't call this function without a registerd donor & therapy record");
   }
 
-  const errors: SubmissionValidationError[] = [];
+  let errors: SubmissionValidationError[] = [];
   const treatment = getTreatment(therapyRecord, mergedDonor, errors);
   if (!treatment) return { errors };
-  checkTreatementHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+  // TODO: Refactor use of push
+  checkTreatmentHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+
+  const {
+    clinicalInfo: { treatment_type },
+    therapies,
+  } = treatment;
+
+  const isRadiationTreatment =
+    Array.isArray(treatment_type) &&
+    treatment_type?.includes('Radiation therapy') &&
+    therapies.some(therapy => therapy.therapyType === 'radiation');
+
+  const radiationErrors: SubmissionValidationError[] = isRadiationTreatment
+    ? radiation(therapyRecord, treatment)
+    : [];
+
+  errors = [...errors, ...radiationErrors];
   return { errors };
 };
 
-export function checkTreatementHasCorrectTypeForTherapy(
+const radiation = (
```

Comment:
this function name could be more descriptive. similar to the other funcs eg. "checkThisThingForThisThing"

Reply:
Yes I've updated to `validateRadiationRecords`

Reply:
So this, I was following the pattern here:
https://github.com/icgc-argo/argo-clinical/blob/master/src/submission/validation-clinical/index.ts#L36

And thinking we should maybe export `radiation` as it's own collection of validators the way we do with other entities. But maybe I'll just leave it as part of `treatment` for now rather than mess with the existing setup.

Reply:
OK so this has changed quite a bit
First, radiation function is now `validateRadiationRecords`
Second, we were previously only validating a single record for a single donor at a time, so more important than separating radiation functions was to create a place for cross file validators
See `src/submission/validation-clinical/therapy.ts @ 70`
This should likely be moved out of therapy into another file but started here